toolchain/lexer/numeric_literal_test.cpp

```diff
@@ -14,6 +14,7 @@
 namespace Carbon::Testing {
 namespace {
 
+using Lex::LexedNumericLiteral;
```

Comment:
Could we put the lexer tests in a `Carbon::Lex::Testing` namespace (or just `Carbon::Lex`; I'm not clear on what the `Testing` namespace adds), and avoid these `using` declarations? It can sometimes make sense to put the tests in a different namespace than the code under test, if the client is expected to be in a different namespace, but that doesn't seem to be the case here.

Reply:
But that's my point: this makes the tests look _unlike_ the calling code, because `LexedNumericLiteral` is only ever used inside the `Lex` namespace, except in the tests.

Reply:
As came up in that thread -- using `Carbon::Lex::Testing` creates a bunch of problems because then any common Carbon testing utilities won't actually be in the namespace at all.

The `Testing` isn't to isolate the test (the anonymous namespace does that), it is to access testing utilities, and those have to live in a single namespace.

Given the tradeoffs, I'm not sure there is a significantly better namespace to use here, and the using declarations seem reasonably minimal?

Reply:
I still don't really understand the tradeoffs being made here, but yeah, I'll take this to Discord.